snuba/datasets/table_storage.py

```diff
@@ -20,12 +20,19 @@
 from snuba.utils.streams.backends.kafka import KafkaPayload
 from snuba.writer import BatchWriter
 
+LOG_APPEND_TIME_STORAGES = {
+    StorageKey.EVENTS,
+    StorageKey.ERRORS,
+    StorageKey.TRANSACTIONS,
+}
+
```

Comment:
I see your problem with the shared topic. 
But why don't we declare a some sort of topic spec configuration, we do not need to put it in settings now (it would probably not even be the right place for most of it) and hide it behind the `build_kafka_topic_spec_from_settings` abstraction ?

`build_kafka_topic_spec_from_settings` provides a `KafkaTopicSpec` object with all the settings to create a topic given the name. Which means it will always return the same value for message timestamp type for the same topic name.

Then we can decide whether to back that method with settings or hardcode it or a mix of the two. Today it would be a mix as the partition number comes from settings while the timestamp type is defined by the application.

Though I would not have a condition based on the storage key to decide the value of the timestamp type parameter. I would have a mapping of topic names to their config values in a module so each storage referencing the `events` topic will get the right settings, there is no need to make this depend on storages.

What do you think? It should not be more complex than the current solution to implement.

Reply:
The problem of just mapping actual topic names to configuration is that is that the topic names themselves are fully configurable. "events" is just the default but a user can change it to whatever else they want via STORAGE_TOPICS. The way I've implemented it, even if the user changed to use a non default topic name instead of "events", the LogAppendTime configuration will still be used because that is defined for those storages. We would lose that we only mapped default topic name to config values and the user wasn't using the default.

Reply:
Right, I guess this is another of those use cases where we should have a better structure for settings, where the application references topics in an abstract way and the settings provide values that are configurable like topic names. 

Let's fix this as it is a bug that is impacting users, would you mind put the topic configuration on the backlog ?

Reply:
Will do 👍